### PR TITLE
Feature/summarization

### PR DESCRIPTION
**Pull request by maurottito**
## 1. What does this PR do?

This PR adds a new bill summarization module and tests.

It introduces logic to:

- Build timeline context from bill steps.
- Rank/filter key legislative events.
- Generate structured Spanish summaries (including edge cases: missing ID, missing bill, no steps).

It also adds unit tests for the summarization 

## 2. How was the functionality tested and verified?
This is a checklist to make sure that the pull request includes complete functionality.

### Code quality
- [ x] All new and existing tests pass locally.
- [ x] My code added tests using pytest for any new functionality.
- [ x] My PR includes documentation updates as relevant (new external commands, updates to options, changes to names, etc.)

### Code review
- [ x] If the changes would break previous functionality, I've communicated this in (1)
- [ x] If there's anything that the reviewer should know, I've added that as a note on the issue or this pull request.

### CODE REVIEW

File: backend/process/summarization.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+MAJORITY_NEEDLES = ("en mayoría", "en mayoria")
+NO_APPROVAL_NEEDLES = ("no aprobación", "no aprobacion", "archivo")
+
+
+def _session_factory():
+    engine = create_engine(settings.DB_URL)
+    return sessionmaker(bind=engine, autocommit=False, autoflush=False)
+
+
+def _enum_text(value) -> str:
```
Comment (cesarnunezh) [inline backend/process/summarization.py:37]: With the changes that I'm doing to `BillStep`, the `step_type` will directly give the value, not the Enum. With that, this function will be no longer needed.

File: backend/process/summarization.py [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+
+    base_scores = {
+        "promulgated": 100,
+        "published": 95,
+        "approved": 85,
+        "vote": 75,
+        "debate": 70,
+        "committee stage": 65,
+        "text update": 60,
+        "presented": 50,
```
Comment (cesarnunezh) [inline backend/process/summarization.py:171]: Maybe I'm not understanding correctly this function and how it impacts the summaries, but usually `presented` are the first step then why `assigned to committee` and `reconsideration` have lower scores? 

Just to have more clarity, did we end up doing this scoring function because there are bills that might go back to early stages? If so, why don't we just filter the most recent per stage?

File: backend/process/summarization.py [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+
+def _paragraph_one(bill_id: str, bill, steps) -> str:
+    start_date = (
+        bill.presentation_date if bill.presentation_date else steps[0].step_date
+    )
+    title = _clean_text(bill.title)
+    title_lower = title.lower()
+    start_month_year = _format_month_year(start_date)
+
+    if "cuestión de confianza" in title_lower:
```
Comment (cesarnunezh) [inline backend/process/summarization.py]: Is there a reason why are we treating different this case?

**Review by cesarnunezh — COMMENTED** [review]
Here are some comments. I haven't try the summarization tool with actual data, maybe we can see that during tomorrow's meeting.

File: backend/process/summarization.py (hunk 1 above)
Comment (maurottito) [reply backend/process/summarization.py:37]: I reviewed that part of your PR, and you are right. But since it is not currently being merged into the dev branch, I'm keeping this for now. I should delete it after your PR is accepted.

File: backend/process/summarization.py (hunk 2 above)
Comment (maurottito) [reply backend/process/summarization.py:171]: Yes, `presented` is usually the first step, and the score is not trying to model chronology but summary relevance: `assigned to committee` and `reconsideration` are usually procedural/supporting events, so they were intentionally set lower than core milestones.  The current ordering also includes date as a tie-breaker, with more recent steps first when scores are equal. 

This scoring was added partly because bills can revisit stages and have repeated events; however, you’re right that a most recent per stage approach is a reasonable alternative and probably clearer. Scoring is mainly useful when `_find_ranked_step`  searches across multiple candidate types (or multiple matches), because it decides which event is returned first; for strict single-type + needle lookups, it has much less effect.

File: backend/process/summarization.py (hunk 3 above)
Comment (maurottito) [reply backend/process/summarization.py]: Not really, I was waiting to have more cases and a bunch of elif statements. I deleted this in the updated version

**Review by cesarnunezh — APPROVED** [review]
